pkg/apis/autoscaling/v1alpha1/cronfederatedhpa_types.go

```diff
@@ -0,0 +1,105 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CronFederatedHPA struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of the CronFederatedHPASpec.
+	// +required
+	Sepc CronFederatedHPASpec `json:"spec"`
+
+	// Status is the current status of the CronFederatedHPA.
+	// +optional
+	Status CronFederatedHPAStatus `json:"status"`
+}
+
+type CronFederatedHPASpec struct {
+	// ScaleTargetRef points to the target resource to scale.
+	// It could be FederatedHPA, Deployment, StatefulSet, or other resource with scale sub-resource.
+	// +required
+	ScaleTargetRef ScaleTargetRef `json:"scaleTargetRef"`
+
+	// Rules is the list of CronFederatedHPARule,
+	// it define when and how to scale the related resources.
+	Rules []CronFederatedHPARule `json:"rules"`
+}
+
+type CronFederatedHPARule struct {
+	// RuleName is the name of the CronFederatedHPARule.
+	// +required
+	RuleName string `json:"ruleName"`
```

Comment:
Indicates which rules are currently supported.

Reply:
Hi, could you please provide more information? Specifically, in this field we are looking for a name which can be any string of characters.